### PR TITLE
chore: Integrating "Clear filter" on remaining listing pages

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
- Follow up to https://github.com/harness/canary/pull/408, this PR integrates "Clear filter" on remaining listing pages.
- Also deletes remaining non-sandbox listing pages.

### CODE REVIEW

File: apps/gitness/src/hooks/useQuery.ts
Comment (abhinavrastogi-harness) [inline apps/gitness/src/hooks/useQuery.ts:1]: let's rename the file as well

File: apps/gitness/src/hooks/useQuery.ts [hunk 1]
```diff
@@ -2,8 +2,16 @@ import { useQueryState } from 'nuqs'
 import { useCallback, useEffect, useState } from 'react'
 import { debounce } from 'lodash-es'
 
-export function useDebouncedQuery(defaultValue = '', delay = 300): [string, (value: string) => void] {
-  const [query, setQuery] = useQueryState('query', { defaultValue })
+export function useDebouncedQueryState({
+  queryKey,
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/hooks/useQuery.ts:6]: this can just be called `key`

File: apps/gitness/src/pages/pipeline/project-pipeline-list.tsx [hunk 2]
```diff
@@ -15,11 +15,11 @@ import { PageResponseHeader } from '../../types'
 import { getExecutionStatus } from '../../utils/execution-utils'
 import { useGetSpaceURLParam } from '../../framework/hooks/useGetSpaceParam'
 import { timeAgoFromEpochTime } from '../pipeline-edit/utils/time-utils'
-import { useDebouncedQuery } from '../../hooks/useQuery'
+import { useDebouncedQueryState } from '../../hooks/useQuery'
 
 export default function ProjectPipelinesPage() {
   const spaceId = useGetSpaceURLParam()
-  const [query, setQuery] = useDebouncedQuery()
+  const [query, setQuery] = useDebouncedQueryState({ queryKey: 'query' })
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/pipeline/project-pipeline-list.tsx:22]: let's follow the same API as `useQueryState`. 
We should be able to call this hook like `useDebouncedQueryState('key', 'value')` and `useDebouncedQueryState('key', { defaultValue: 'value' })`

File: apps/gitness/src/pages/project-settings/project-settings-members-page.tsx [hunk 3]
```diff
@@ -25,41 +27,36 @@ import { timeAgoFromEpochTime } from '../pipeline-edit/utils/time-utils'
...
   const [dialogState, setDialogState] = useState<DialogState>({
     isDialogDeleteOpen: false,
     selectedMember: null
   })
-
-  const { sort, query: currentQuery } = useCommonFilter<MembershipListQueryQueryParams['sort']>()
-  const [query, setQuery] = useQueryState('query', {
-    defaultValue: currentQuery || ''
-  })
+  const { sort } = useCommonFilter<MembershipListQueryQueryParams['sort']>()
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/project-settings/project-settings-members-page.tsx:43]: we're still using `useCommonFilter`? Will this be removed in a separate PR?

Comment (vardanbansal-harness) [reply apps/gitness/src/pages/project-settings/project-settings-members-page.tsx:43]: This is still used for `sort` in few pages. Will remove it separately.

File: apps/gitness/src/pages/pipeline/project-pipeline-list.tsx (hunk 2 above)
Comment (vardanbansal-harness) [reply apps/gitness/src/pages/pipeline/project-pipeline-list.tsx]: Done

File: apps/gitness/src/hooks/useQuery.ts (hunk 1 above)
Comment (vardanbansal-harness) [reply apps/gitness/src/hooks/useQuery.ts]: Done

File: apps/gitness/src/hooks/useQuery.ts [hunk 4]
```diff
@@ -2,8 +2,16 @@ import { useQueryState } from 'nuqs'
...
+  defaultValue = '',
+  delay = 300
+}: {
+  queryKey: string
+  defaultValue?: string
+  delay?: number
```
[the remark this answers is not in the document]
Comment (vardanbansal-harness) [reply apps/gitness/src/hooks/useQuery.ts]: Done

File: apps/gitness/src/hooks/useQuery.ts
Comment (vardanbansal-harness) [reply apps/gitness/src/hooks/useQuery.ts:1]: Done

**Review by abhinavrastogi-harness — APPROVED** [review]
